docs/docs/guides/events_subscriptions/custom_subscriptions.md

```diff
@@ -0,0 +1,230 @@
+---
+sidebar_position: 2
+sidebar_label: 'Custom Subscriptions'
+---
+
+# Custom Subscription
+
+You can extend the `Web3Subscription` class to create custom subscriptions. This way you can subscribe to custom events emitted by the provider.
+
+:::note
+This guid is most likely for advanced users who are connecting to a node that provide additional custom subscription. And for normal users, the standard subscriptions are supported out of the box as you can find in [Supported Subscriptions](/guides/events_subscriptions/supported_subscriptions).
+:::
+
+:::important
+If you are the developer who provide custom subscriptions to users. We encourage you to develop a web3.js Plugin after you go through the guide below. However, you can find how to develop a plugin at [web3.js Plugin Developer Guide](/guides/web3_plugin_guide/plugin_authors).
+
+And even if you are not the developer how provider this custom subscriptions, we encourage you to write a web3.js plugin for the custom subscription, and publish it to the npm registry. This way you can help the community. And they might contribute to your repository helping for things like: feature addition, maintenance, and bug detection.
+:::
+
+## Implementing the Subscription
+
+### Extending `Web3Subscription`
+
+To create a custom subscription, start by extending the `Web3Subscription` class. However, `Web3Subscription` is generically typed. And, generally, you need only to provide the first two types which are:
+
+-   `EventMap` - The event map for events emitted by the subscription
+-   `ArgsType` - The arguments passed to the subscription
+
+For example:
+
+```ts
+class MyCustomSubscription extends Web3Subscription<
+	{
+		// here provide the type of the `data` that will be emitted by the node
+		data: string,
+	},
+	// here specify the types of the arguments that will be passed to the node when subscribing
+	{
+		customArg: string,
+	},
+> {
+	// ...
+}
+```
+
+### Specify the Subscription Arguments
+
+You need to specify the exact data that will be passed to the provide. You do this by overriding `_buildSubscriptionParams` in your class. It could be something as follow:
```

Comment:
```suggestion
You need to specify the exact data that will be passed to the provider. You do this by overriding `_buildSubscriptionParams` in your class. It could be something as follow:
```

do you want to mention provider or provide function, its not clear??